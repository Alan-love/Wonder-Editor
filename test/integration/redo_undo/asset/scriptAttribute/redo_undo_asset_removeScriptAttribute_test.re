open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: asset remove script attribute", () => {
    let sandbox = getSandboxDefaultVal();

    let _simulateOnceRemoveScriptAttribute = () => {
      let assetTreeData =
        MainEditorAssetTreeTool.BuildAssetTree.buildEmptyAssetTree();
      let addedScriptAttributeNodeId1 = MainEditorAssetIdTool.getNewAssetId();

      MainEditorAssetHeaderOperateNodeTool.addScriptAttribute();
      MainEditorAssetHeaderOperateNodeTool.removeScriptAttributeNode(
        ~scriptAttributeNodeId=addedScriptAttributeNodeId1,
        (),
      );
    };

    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initState(~sandbox, ());
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode,
      );
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test undo operate", () =>
      describe("test undo one step", () =>
        test("step which from first to zero", () => {
          _simulateOnceRemoveScriptAttribute();

          RedoUndoTool.undoHistoryState();

          BuildComponentTool.buildAssetChildrenNode()
          |> ReactTestTool.createSnapshotAndMatch;
        })
      )
    );

    describe("test redo operate", () =>
      describe("test redo one step", () =>
        test(
          "undo step which from first to zero, redo step which from zero to first",
          () => {
          _simulateOnceRemoveScriptAttribute();

          RedoUndoTool.undoHistoryState();
          RedoUndoTool.redoHistoryState();

          BuildComponentTool.buildAssetChildrenNode()
          |> ReactTestTool.createSnapshotAndMatch;
        })
      )
    );
  });