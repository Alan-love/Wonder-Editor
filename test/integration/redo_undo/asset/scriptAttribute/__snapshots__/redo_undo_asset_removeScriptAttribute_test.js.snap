// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: asset remove script attribute test redo operate test redo one step undo step which from first to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-asset-assetChildren"
/>
`;

exports[`redo_undo: asset remove script attribute test undo operate test undo one step step which from first to zero 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/scriptAttribute.png"
      />
    </div>
    <div
      className="item-text item-active"
    >
      <span>
        New Script Attribute
      </span>
    </div>
  </article>
</article>
`;
